Data Structures and Algorithms/Data Structures and Algorithms/Queue/QueueRingBuffer.swift

```diff
@@ -38,3 +44,12 @@ extension QueueRingBuffer: CustomStringConvertible {
         return String(describing: buffer)
     }
 }
+
+extension QueueRingBuffer {
+    public mutating func reversed(size: Int) -> QueueRingBuffer {
+        let bufferReversed = self.buffer.reversed()
```

Comment:
for reversing the ring buffer you need to take account for where the read index is, the reversed() call you are doing will reverse all the buffer from index 0.